app/src/main/java/org/schabi/newpipe/player/BackgroundPlayer.java

```diff
@@ -633,5 +641,48 @@ public void onCompleted() {
             updateNotification(R.drawable.ic_replay_white);
             lockManager.releaseWifiAndCpu();
         }
+
+        @Override
+        public void setTimer(int hourOfDay, int minute) {
+            long time = ((hourOfDay * 60) + minute) * 60 * 1000;
+            if (time == 0) {
+                Toast.makeText(getApplicationContext(), getString(R.string.timer_select_time_message), Toast.LENGTH_SHORT).show();
+                return;
+            }
+            timer.cancel();
+            timer.purge();
+            timerHour = hourOfDay;
+            timerMinute = minute;
+            timer = new Timer();
+            timer.schedule(new TimerTask() {
+                @Override
+                public void run() {
+                    new Handler(Looper.getMainLooper()).post(() -> {
+                        if (DEBUG) Log.d(TAG, getString(R.string.log_stopping_the_player));
```

Comment:
This is a debug string to be printed in the logcat console, it needs to be in English language and it should be hardcoded.
```suggestion
                        if (DEBUG) Log.d(TAG, "Timer went off: stopping the player");
```

Reply:
I changed this as @TobiGr requested the changes for same. I'll change it back to hard coded.

Reply:
TobiGr requested changes for strings visible to the user, not for the logging strings, so please change this ;-)